### ISSUE TITLE
C++11 Lambda的一个坑

### ISSUE THREAD

**Issue by coolexp** [coolexp/coolexp.github.com#5]
在类内如果书写这样的方法,在类析构时会产生坑：
有类成员 m_pVector;// 容器vector
<code>
void 类::test(){
  std::thread t([=](){
       std::this_thread::sleep_for(std::chrono::milliseconds(2000));
       m_pVector.clear();// crash
   });
   t.detach();
}
</code>
